common/util.c

```diff
@@ -23,7 +23,7 @@ int numlen(int n) {
 	if (n == 0) {
 		return 1;
 	}
-	return log10(n) + 1;
+	return log10(abs(n)) + 1;
```

Comment:
Now that we're not calling this with -1 again, could you either revert this change or reject negative numbers until we have a use case for it?

Reply:
I think it should remain like this. Negative numbers have lengths too, and before this change it would output a large negative number which could trigger a bug.

Reply:
In that case, I think for negative numbers it should add an extra character to account for the '-' at the start.